pkg/services/baremetal/host/host.go

```diff
@@ -592,46 +595,18 @@ func (s *Service) actionRegistering(_ context.Context) actionResult {
 	}
 	sshClient := s.scope.SSHClientFactory.NewClient(in)
 
-	// Check hostname with sshClient
-	out := sshClient.GetHostName()
-	hostName := trimLineBreak(out.StdOut)
-	if hostName != rescue {
-		// give the reboot some time until it takes effect
-		if s.hasJustRebooted() {
-			return actionContinue{delay: 2 * time.Second}
-		}
-
-		isSSHTimeoutError, isSSHConnectionRefusedError, err := s.analyzeSSHOutputRegistering(out)
-		if err != nil {
-			// This can happen if the bare-metal server was taken by another mgt-cluster.
-			// Check in https://robot.hetzner.com/server for the "History" of the server.
-			return actionError{err: fmt.Errorf("failed to handle incomplete boot - registering: %w", err)}
-		}
-
-		failed, err := s.handleIncompleteBoot(true, isSSHTimeoutError, isSSHConnectionRefusedError)
-		if failed {
-			return s.recordActionFailure(infrav1.PermanentError, err.Error())
-		}
-		if err != nil {
-			return actionError{err: fmt.Errorf(errMsgFailedHandlingIncompleteBoot, err)}
-		}
-		timeSinceReboot := "unknown"
-		if s.scope.HetznerBareMetalHost.Spec.Status.LastUpdated != nil {
-			timeSinceReboot = time.Since(s.scope.HetznerBareMetalHost.Spec.Status.LastUpdated.Time).String()
-		}
-
-		s.scope.Logger.Info("Could not reach rescue system. Will retry some seconds later.", "out", out.String(), "hostName", hostName,
-			"isSSHTimeoutError", isSSHTimeoutError, "isSSHConnectionRefusedError", isSSHConnectionRefusedError, "timeSinceReboot", timeSinceReboot)
-		return actionContinue{delay: 10 * time.Second}
+	ok, res := s.validateRescueSystemIsActive(ctx, sshClient)
```

Comment:
can you maybe keep the original code here? Because you don't re-use the new utility function anymore, as you probably intended originally. If you don't re-use, I'd keep it as is

Reply:
ok, I inlined the function which was used only once: 3e6fd3e4